certbot/certbot/compat/filesystem.py

```diff
@@ -76,6 +78,35 @@ def umask(mask: int) -> int:
     return previous_umask
 
 
+class TempUmask(object):
```

Comment:
What do you think about using https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager to remove the need for the custom class?

If you don't like that approach for some reason, what about making this class private? This module is technically part of our public API (since it's not under `_internal`) and I see no reason anyone should ever directly use this class.

Reply:
i didn't know about this! definitely a cleaner approach, thanks